docs/source/concepts/bento.rst

```diff
@@ -796,76 +796,70 @@ will build a new image on top of the base_image with the following steps:
         bentoml containerize iris_classifier:latest --platform=linux/amd64
 
 
-Docker Template (Danger Zone)
-"""""""""""""""""""""""""""""
+Dockerfile Template (Beta)
+""""""""""""""""""""""""""
 
-The :code:`docker_template` field gives user the full control over how the
-:code:`Dockerfile` was generated in a Bento. Users can use this field to
-customize Bento's Dockerfile instruction set to suits their needs.
+The :code:`dockerfile_template` field gives user the full control over how the
+:code:`Dockerfile` was generated in a Bento. Users can extend the template used for
+generating Dockerfile in a Bento, to suits their needs.
 
-First, create a :code:`Dockerfile.template` file next to your :code:`bentofile.yaml` build file.
-This template file is a mixed between a :code:`Dockerfile` and a :code:`Jinja` template file:
+First, create a :code:`Dockerfile.template` file next to your :code:`bentofile.yaml`
+build file. This file should follow the
+`Jinja2 <https://jinja.palletsprojects.com/en/3.1.x/>`_ template language, and extend
+BentoML's base template and blocks. The template should render a valid
+`Dockerfile https://docs.docker.com/engine/reference/builder/`_. For example:
```

Comment:
Should we call it `Dockerfile.j2`?

Seems like it doesn't need to be saved alongside the `bentofile.yaml`?

```suggestion
First, create a `Jinja2 <https://jinja.palletsprojects.com/en/3.1.x/>`_ template for your Dockerfile.
This file can extend BentoML's base template and blocks. For example:
```

Reply:
the reason for L806 is that we want to enforce this practice to our users. Eventhough we accept the any path to templates, we should just mention to them that they should put it in the build context.

Reply:
Ah, I see. We should just make it clear it's a recommendation, then; I can see use-cases where people want to have multiple bentos use the same dockerfile template.